git-branchless/src/commands/hide.rs

```diff
@@ -89,14 +113,62 @@ pub fn hide(effects: &Effects, hashes: Vec<String>, recursive: bool) -> eyre::Re
         }
     }
 
+    if delete_branches {
+        // Delete the abandonded branches by "moving" them from their current OID to a Zero OID.
+        let abandoned_branches = abandoned_branches
+            .values()
+            .map(|oid| (*oid, MaybeZeroOid::Zero))
+            .collect();
```

Comment:
I think it should be safe to pass OIDs which don't point to a branch to `move_branches`. Then we can reduce the scope of `abandoned_branches` by moving its construction below and writing something like `commits.iter().map(|commit| (commit.get_oid(), MaybeZeroOid::Zero)).collect()` here.